### PR TITLE
Add aten::pow operator and layer tests

### PR DESCRIPTION
**Pull request by sikorsl1**
**Details:**

-  `aten::pow` operator has been added
- layer test for this operator has been added

**Ticket:**

- 94282

### CODE REVIEW

File: tests/layer_tests/pytorch_tests/test_pow.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    """
+
+    def _prepare_input(self):
+        return self.test_input
+
+    def create_model(self):
+        class aten_pow(torch.nn.Module):
+
+            def forward(self, input_data, exponent):
+                return torch.pow(input_data, exponent)
```
Comment (slyalin) [inline tests/layer_tests/pytorch_tests/test_pow.py:33]: Looks like this only line will change from test to test. Could we implement test in more compact manner?

Comment (slyalin) [reply tests/layer_tests/pytorch_tests/test_pow.py:33]: I mean all tests.

Comment (mvafin) [reply tests/layer_tests/pytorch_tests/test_pow.py:33]: Not really just one line. For more complicated ops you would need to modify whole model. E.g. `__init__` method. But I definitely welcome any improvements here. Please also note that I will add verification of the fact that op kind exist in model, so test implementer will also write op kind in the test

Comment (slyalin) [reply tests/layer_tests/pytorch_tests/test_pow.py:33]: In this case we will have two lines of code: one in `__init__` and one in `forward`.

File: tests/layer_tests/pytorch_tests/test_pow.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+@pytest.mark.parametrize('test_input', [(np.array([[1, 2], [3, 4]], dtype=np.float32),
+                                         np.array([[1, 1], [2, 2]], dtype=np.float32),),
+                                        (np.array([[1, 2], [3, 4]], dtype=np.float32),
+                                         np.array([2, 3], dtype=np.float32),),
+                                        (np.array([[1, 2], [3, 4]], dtype=np.float32),
+                                         np.array([2], dtype=np.float32),),
+                                        (np.array([5, 6], dtype=np.float32),
+                                         np.array([[1, 2], [3, 4]], dtype=np.float32),),
+                                        (np.array([5], dtype=np.float32),
+                                         np.array([[1, 2], [3, 4]], dtype=np.float32),)])
```
Comment (slyalin) [inline tests/layer_tests/pytorch_tests/test_pow.py:19]: When we cover more data types (if any, fp16 for example, I don't know), will we just double this block? Looks like for all operations that support broadcasting we will do this "dimensional exercise". Can we provide some helpers to produce less boilerplate code in each test? @mvafin?

Comment (mvafin) [reply tests/layer_tests/pytorch_tests/test_pow.py:19]: I don't think we should double this block with additional types. We want to check translation correctness, not correctness of openvino Power operation

**Review by mvafin — APPROVED** [review]
Looks good to me

**Pull request merged**